### PR TITLE
Add ERC20 support to getSwaps

### LINKED ISSUE #950: Fully support getSwaps in Ethereum connection
Currently our Ethereum connection's `getSwaps` implementation supports querying by ID for both ETH and ERC20 token swaps because the prefix will tell the connection which contract to query. For other types of query we don’t have this information. Options here include:
1. A flag to indicate which type of swap is desired, possibly by way of providing an ERC20 contract address to further filter results.
1. Getting both types of swaps and merging them together.
1. Adding additional view functions to the smart contracts themselves.
1. Separate `EthereumConnection` instances for Ether/ERC20 or even each ERC20 token.

### PR DESCRIPTION
**Pull request by willclarktech**
Closes #950

### CODE REVIEW

File: packages/iov-ethereum/src/ethereumconnection.ts [hunk 1]
```diff
@@ -138,6 +155,130 @@ export class EthereumConnection implements AtomicSwapConnection {
...
+    const swapIdBegin = 0;
+    const swapIdEnd = swapIdBegin + 32;
+
+    return {
+      kind: SwapProcessState.Aborted,
+      swapIdBytes: bytes.slice(swapIdBegin, swapIdEnd) as SwapIdBytes,
+    };
+  }
+
+  private static updateSwapInList(
```
Comment (webmaster128) [inline packages/iov-ethereum/src/ethereumconnection.ts]: It looks like this does the same as BCP's AtomicSwapMerger. Could you have a look at that?

File: packages/iov-ethereum/src/ethereumconnection.ts [hunk 2]
```diff
@@ -138,6 +155,130 @@ export class EthereumConnection implements AtomicSwapConnection {
...
+    const recipientBegin = senderEnd;
+    const recipientEnd = recipientBegin + 32;
+    const hashBegin = recipientEnd;
+    const hashEnd = hashBegin + 32;
+    const amountBegin = hashEnd;
+    const amountEnd = amountBegin + 32;
+    const timeoutBegin = amountEnd;
+    const timeoutEnd = timeoutBegin + 32;
+    const erc20ContractAddressBegin = timeoutEnd;
+    const erc20ContractAddressEnd = erc20ContractAddressBegin + 32;
```
[the remark this answers is not in the document]
Comment (willclarktech) [reply packages/iov-ethereum/src/ethereumconnection.ts]: This is the contract where the token is defined. There's no equivalent for ETH. In that case, the returned data is just shorter.

File: packages/iov-ethereum/src/ethereumconnection.ts (hunk 1 above)
Comment (willclarktech) [reply packages/iov-ethereum/src/ethereumconnection.ts]: So the difference is that `AtomicSwapMerger` is merging full transactions into a list of swaps, whereas what's happening here is that events from the smart contract are being merged into a list of swaps. Claimed/Aborted events have very minimal information compared to transactions.

Comment (webmaster128) [reply packages/iov-ethereum/src/ethereumconnection.ts]: Perfect, thanks for clarifying

File: packages/iov-ethereum/src/ethereumconnection.ts [hunk 3]
```diff
@@ -138,6 +155,130 @@ export class EthereumConnection implements AtomicSwapConnection {
...
+
+    return {
+      kind: SwapProcessState.Open,
+      data: {
+        id: {
+          prefix: prefix,
+          data: bytes.slice(swapIdBegin, swapIdEnd) as SwapIdBytes,
+        },
+        sender: toChecksummedAddress(Abi.decodeAddress(bytes.slice(senderBegin, senderEnd))),
+        recipient: toChecksummedAddress(Abi.decodeAddress(bytes.slice(recipientBegin, recipientEnd))),
```
[the remark this answers is not in the document]
Comment (webmaster128) [reply packages/iov-ethereum/src/ethereumconnection.ts]: Could you rebase and update this now that #959 is in?

**Pull request merged**